backend/src/main/java/net/vidux/camhub/discovery/DiscoveryService.java

```diff
@@ -0,0 +1,43 @@
+package net.vidux.camhub.discovery;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+
+@Component
+public class DiscoveryService {
+  protected static Logger log = LoggerFactory.getLogger(DiscoveryService.class.getName());
+
+  CompletableFuture<Set<RawCameraData>> cameraScanTask;
+
+  @Autowired private RawCameraEventPublisher rawCameraPublisher;
+
+  @Autowired private CameraScan cameraScan;
+
+  public void requestDiscovery() throws DiscoveryException {
+    if (cameraScanTask != null && !cameraScanTask.isDone()) {
+      throw new DiscoveryException("Camera Scan task is already running");
+    }
+    discover();
+  }
```

Comment:
Ha az a feladat, hogy egy időben egyetlen camera scan task lehet egy alkalmazáson belül, akkor rendben, viszont szerintem érdemes lenne valamilyen módon lekérdezhetővé tenni, hogy fut-e épp egy camera scan.

Az lenne a szép szerintem ha úgy működne, hogy ha valaki megnyitja az oldalt, akkor valamilyen módon látná, hogy fut-e a scannelés, mondjuk a scan gomb inaktív lenne és egy animált ikon jelezné, hogy a scannelés fut és mondjuk azt is lehetne látni, hogy ki indította a scannelést.
Amikor befejeződött a scannelés, akkor pedig aktívvá válna a gomb mindenki számára.
Erre egy megoldás lehet az SSE (server-sent events) használata, amit szerintem meg lehet oldani úgy, hogy a kliens feliratkozik a scannelés eseményre és amikor a szerveren befejeződött a scan task, akkor az küldene egy event-et a kliensnek.
Nem azt mondom, hogy így legyen megoldva, hiszen nem ez a feladat, de egy ilyet is el tudnék képzelni a jövőben, ezért szerintem a scan állapotát érdemes lenne lekérdezhetővé tenni, tehát én a feltételt kiemelném egy külön metódusba (pl.: isCameraScanRunning).

A discover metódus pedig csak akkor indítana scan-t, ha az `isCameraScanRunning == false`. 
Így nincs szükség szerintem kivétel dobásra abban az esetben, ha épp fut egy scan, hiszen ez nem egy kivételes eset, hanem egy állapot.

Reply:
Konkrétan az volt az elképzelés, hogy a usernek nem fontos, hogy ki és mikor indított keresést, hanem az, hogy ha megnyomja a gombot kapjon friss infókat. Esetleg egy jövőbeli tasknak ez jó lehet, de időközben az is kiderült, hogy ez a task sokkal gyorsabban fut le mint gondoltuk (<1mp).
A custom exception és a kivétel dobás meg itt külön kérés volt, de lehet nem lesz szükség rá, mert nem fogunk vele semmit se kezdeni.

Reply:
A kiemelést megcsináltam, hogy ha esetleg késöbb mégis szeretnénk ilyesmit ehhez már nagyon ne kelljen nyúlni